python/create-onchain-agent/create_onchain_agent/cli.py

```diff
@@ -131,56 +158,95 @@ def create_project():
     else:
         package_name = suggested_package_name
 
-    # Select network using arrow keys
-    # console.print("\n[cyan]Select a network:[/cyan]")
-    
-    network_name = questionary.select(
-        "Choose a network network:",
-        choices=[name for name, _ in NETWORK_CHOICES],
-        default="Base Sepolia (default)",
-        style=custom_style  # Apply custom styling
+    # Choose network type
+    network_type = questionary.select(
+        "Choose network type:",
+        choices=[
+            "Mainnet",
+            "Testnet",
+            "Custom Chain ID",
+        ],
+        style=custom_style
     ).ask()
 
-    # Map selection to network key
-    network = next(n for n in NETWORK_CHOICES if n[0] == network_name)[1]
+    network = None
+    chain_id = None
+    rpc_url = None
 
-    # If "Other" is selected, prompt for EVM Chain ID
-    if network == "other":
-        network = questionary.text(
-            "Enter the EVM Chain ID for your custom network:",
+    if network_type == "Custom Chain ID":
+        # Handle custom EVM network
+        chain_id = questionary.text(
+            "Enter your chain ID:",
+            validate=lambda text: text.strip().isdigit() or "Chain ID must be a number",
             style=custom_style
-        ).ask().strip()
-
-    # Determine wallet provider
-    if network in CDP_SUPPORTED_NETWORKS:
-        wallet_provider = questionary.select(
-            "Select a wallet provider:",
-            choices=["CDP Wallet Provider", "Ethereum Account Wallet Provider"],
-            default="CDP Wallet Provider",
-            style=custom_style  # Apply custom styling
         ).ask()
 
-        wallet_provider = "cdp" if wallet_provider.startswith("CDP") else "eth"
+        rpc_url = questionary.text(
+            "Enter your RPC URL:",
+            validate=lambda text: (
+                text.strip().startswith(("http://", "https://")) or 
+                "RPC URL must start with http:// or https://"
```

Comment:
nit: we should probably only allow URLs starting with `https://`

Reply:
I think http should be allowed due to developers running local nodes. If anyone is trying to connect to ganache or equivalent, they'll be putting in `http://127.0.0.1:7545 ` as the url